03week/towersOfHanoi.js

```diff
@@ -19,23 +24,67 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
+function movePiece(startStack, endStack) {
+     if(startStack === 'a' && (endStack === 'b' || endStack === 'c')) {
+       return true;
+     } else if (startStack === 'b' && (endStack === 'a' || endStack === 'c')) {
+       return true;
+     } else if (startStack === 'c' && (endStack === 'a' || endStack === 'b')) {
+       return true;
+     }
+}
 
+const isValid = (startStack, endStack) => {
+  try {
+    if(startStack === '' || endStack === '') {
+      throw 'Please enter a, b, or c';
+    }
+    if((startStack !== 'a') && (endStack !== 'b' || endStack !== 'c')) {
+      throw 'Please enter a valid letter.  Please don\'t enter the same letter.';
+    }
+    if((startStack !== 'b') && (endStack !== 'a' || endStack !== 'c')) {
+      throw 'Please enter a valid letter.  Please don\'t enter the same letter.';
+    }
+    if((startStack !== 'c') && (endStack !== 'a' || endStack !== 'b')) {
+      throw 'Please enter a valid letter.  Please don\'t enter the same letter.';
+    }
+  } catch(err) {
+    console.log(err);
+  }
 }
 
-function isLegal() {
+function isLegal(startStack, endStack) {
   // Your code here
-
+  try {
+    if(stacks.b[stacks[startStack]] < stacks.c[stacks[endStack]] || stacks.b.length == 0 || stacks.c.length == 0) {
+      throw 'Illegal move!  You can\'t place a bigger tile on top of a smaller tile.  Please choose another move.';
+      return false;
+    }
+  } catch(err) {
+    console.log(err);
+  }
 }
 
 function checkForWin() {
   // Your code here
-
+  if(stacks.b.length === 4 || stacks.c.length === 4) {
+    return true;
+  } else {
+    return false;
+  }
 }
 
 function towersOfHanoi(startStack, endStack) {
   // Your code here
+    if(movePiece(startStack, endStack)) {
+      return stacks[endStack].push(stacks[startStack].pop());
+    }
+
+    if(checkForWin()) {
```

Comment:
This is saying, if there isn't a win, check for move validity. It should be the other way around. You should be checking if the move is legal and valid and if it is then check for a win.